common/djangoapps/student/tasks.py

```diff
@@ -0,0 +1,50 @@
+"""
+This file contains celery tasks for sending email
+"""
+import time
+from django.conf import settings
+from django.core import mail
+
+from celery.task import task  # pylint: disable=no-name-in-module, import-error
+from celery.utils.log import get_task_logger  # pylint: disable=no-name-in-module, import-error
+
+log = get_task_logger(__name__)
+
+
+@task()
+def send_activation_email(user, subject, message, from_address):
+    """
+        Sending an activation email to the users.
+    """
+    retries = settings.RETRY_ACTIVATION_EMAIL_MAX_ATTEMPTS
```

Comment:
Just for a suggestion, wouldn't it be more readable if we just compare if our retries didn't acceded the maximum retries? 

In other words what I am saying rather than logging 
`"Retrying sending email, attempt # 5..."`
`"Retrying sending email, attempt # 4..."`

Make it such

```
max_retires = settings.RETRY_ACTIVATION_EMAIL_MAX_ATTEMPTS
retries = 0

while retries < max_retires:
    ...
    ...
    except Exception: 
        retries +=1
        if retries >= max_retries:
             log.info("Activataion Email has been sent to User ....")
             ....
       else:
           log.info("Retrying sending email, attempt #")  #now this would log correct attempts, 1, 2, 3, ... 


```
Well its just a suggestion, its upto you, :-)